torch/csrc/distributed/c10d/ProcessGroupGloo.cpp

```diff
@@ -3150,6 +2375,92 @@ void ProcessGroupGloo::enableCollectivesTiming() {
   // Nothing to do to enable timing
 }
 
+void initializeStreamsEvents(
+    const std::vector<at::Tensor>& tensors,
+    std::vector<c10::Stream>& streams,
+    std::vector<c10::Event>& events) {
```

Comment:
Any reason why we can use c10::Event and c10::Stream for gloo? Oh, maybe c10::Event only works with c10::Stream?

Reply:
This was existing code -- I'm not sure on that, I just moved this